Sources/NIO/Linux.swift

```diff
@@ -132,5 +136,60 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func contentsOfFile(path: String) throws -> Substring {
+        let fh = try NIOFileHandle(path: path)
+        defer { try? fh.close() }
+        // linux doesn't properly report /sys/fs/cgroup/* files lengths so we use a reasonable limit
+        let buffSize = 1024
+        var buf = ByteBufferAllocator().buffer(capacity: buffSize)
+        try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: buffSize) { ptr in
+            let res = try fh.withUnsafeFileDescriptor { fd -> IOResult<ssize_t> in
+                return try Posix.read(descriptor: fd, pointer: ptr.baseAddress!, size: buffSize)
```

Comment:
instead of `buffSize` you should use `ptr.count`. `ByteBufferAllocator` reserves the right to allocate more so if it does, we can make use of that :).